src/jit/lowerxarch.cpp

```diff
@@ -3519,6 +3537,8 @@ void Lowering::TreeNodeInfoInitCmp(GenTreePtr tree)
                             castOp1->gtIntCon.gtIconVal = (UINT8)castOp1->gtIntCon.gtIconVal;
                         }
 
+                        // We're within "if (CheckImmedAndMakeContained(tree, op2))", so isn't the following condition
+                        // always true?
```

Comment:
If you're not going to change this, you should at least make this a TODO-Cleanup